frontend/lib/pages/item.dart

```diff
@@ -214,4 +282,77 @@ class ItemDetailsScreen extends StatelessWidget {
       ),
     );
   }
+
+  void fetchProduct() async {
+    print('fetchProduct called');
+    const url = "http://192.168.0.2:8080/api/v1/product/1"; // Replace with the device's IPv4 address (the one that's hosting the backend)
+    final uri = Uri.parse(url);
+
+    try {
+      final response = await http.get(uri);
+
+      if (response.statusCode == 200) {
+        final body = response.body;
+        final json = jsonDecode(body);
+
+        setState(() {
+          productData = json;
+          userName = productData?["sellerName"] ?? '';
+          fetchUser(userName);
+        });
+
+        print('fetchProduct completed');
+      } else {
+        // Handle non-200 HTTP status codes (e.g., server errors)
+        print('Error: HTTP Status Code ${response.statusCode}');
+      }
+    } catch (e) {
+      // Handle network-related errors (e.g., SocketException, HttpException)
+      print('Error: $e');
+    }
+  } // function fetchProduct
+
+  void fetchUser(String? userName) async {
+    print('fetchUser called');
+    final url = "http://192.168.0.2:8080/api/v1/user/info/?username=$userName"; // Replace with the device's IPv4 address (the one that's hosting the backend)
```

Comment:
1. we can change the URL to "http://192.168.0.2:8080/api/" to be a constant but not a big issue (just that if we have to change for every single time we can it becomes an issue since there will be inevitably some that we miss out along the way) 
2. The entire thing looks good. we can call once and get all the respective fields that we require instead of calling it for every field

Reply:
but no need to worry we can look into it later